### ISSUE TITLE
Can Dist SQL completely replace configuration files?

### ISSUE THREAD

**Issue by CrackerSuperior** [apache/shardingsphere#12735]

**Comment by strongduanmu** [comment]
@CrackerSuperior Yes, you can use distsql to manage your configuration without any other files.

**Comment by CrackerSuperior** [comment]
> @CrackerSuperior Yes, you can use distsql to manage your configuration without any other files.

For example: If I wanted to make default sharding rules for all tables in my library, what would I do?

**Comment by strongduanmu** [comment]
Currently, distsql is used to configure sharding and only `autoTables` is supported. More configuration methods will be supported later.

**Comment by strongduanmu** [comment]
@CrackerSuperior You mean default sharding strategy? Currently distsql only supports autoTable sharding.

**Comment by CrackerSuperior** [comment]
> @CrackerSuperior您的意思是默认分片策略？目前 distsql 仅支持 autoTable 分片。

Yes, to replace defaultTableStrategy and defaultDatabaseStrategy in the configuration file.

**Comment by CrackerSuperior** [comment]
> @CrackerSuperior You mean default sharding strategy? Currently distsql only supports autoTable sharding.

If I use the configuration file method, do I need to configure a new configuration file every time I add a new logic library, and restart my sharding to let it load the new configuration file?

**Comment by RaigorJiang** [comment]
> If I use the configuration file method, do I need to configure a new configuration file every time I add a new logic library, and restart my sharding to let it load the new configuration file?

It does need to be restarted, so ShardingSphere designed DistSQL to make the configuration take effect in real time.

**Comment by RaigorJiang** [comment]
> For example: If I wanted to make default sharding rules for all tables in my library, what would I do?

This is a good demand and I believe it will be provided in the future. You can also improve it together.

**Comment by RaigorJiang** [comment]
Hi @CrackerSuperior 
DistSQL has been updated and can now fully support the configuration ways provided by the configuration file. Welcome to try.
https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-sharding-rule/

**Comment by CrackerSuperior** [comment]
> Hi @CrackerSuperior DistSQL has been updated and can now fully support the configuration ways provided by the configuration file. Welcome to try. https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/usage/distsql/syntax/rdl/rdl-sharding-rule/

OK, thanks
